CRFModuleValidation/MasterScheduledActivityManager.swift

```diff
@@ -0,0 +1,563 @@
+//
+//        let dispatchGroup = DispatchGroup()
+//
+//        let archiveResult = SBAActivityResult(taskIdentifier: identifier, taskRun: UUID(), outputDirectory: nil)
+//        archiveResult.schedule = schedule
+//        archiveResult.schemaRevision = self.bridgeInfo.schemaReferenceWithIdentifier(identifier)?.schemaRevision ?? 1
+//
+//        let heartRateResult = HKSamplesDataResult(identifier: "heartRate")
+//        let pedometerResult = PedometerDataResult(identifier: "pedometer")
+//
+//        // Get heart rate samples
+//        let datePredicate = HKQuery.predicateForSamples(withStart: startDate, end: endDate, options: .strictStartDate)
+//        let healthStore = HKHealthStore()
+//        dispatchGroup.enter()
+//        let heartRateQuery = HKAnchoredObjectQuery(type: HKSampleType.quantityType(forIdentifier: .heartRate)!,
+//                                           predicate: datePredicate,
+//                                           anchor: nil,
+//                                           limit: HKObjectQueryNoLimit) { (_, samples, _, _, error) in
+//                                            heartRateResult.samples = samples as? [HKQuantitySample]
+//                                            heartRateResult.permissionDenied = (error != nil)
+//                                            dispatchGroup.leave()
+//        }
+//        healthStore.execute(heartRateQuery)
+//
+//        // Get pedometer per day
+//        if SBAPermissionsManager.shared.isPermissionGranted(for: SBAPermissionObjectType(permissionType: .coremotion)) &&
+//           CMPedometer.isStepCountingAvailable() {
+//            let collector = PedometerDataCollector()
+//            dispatchGroup.enter()
+//            collector.queryPedometer(from: startDate, to: endDate, withHandler: { (pedometerData) in
+//                pedometerResult.pedometerData = pedometerData
+//                dispatchGroup.leave()
+//            })
+//        }
+//        else {
+//            pedometerResult.permissionDenied = true
+//        }
+//
+//        dispatchGroup.notify(queue: DispatchQueue.global(qos: .background)) {
+//            let stepResult = ORKStepResult(stepIdentifier: identifier, results: [heartRateResult, pedometerResult])
+//            archiveResult.results = [stepResult]
+//            if let archive = self.archive(for: archiveResult) {
+//
+//                // Update the schedule
+//                // Note: the client data is set during the sendUpdated() call
+//                schedule.startedOn = Date()
+//                schedule.finishedOn = Date()
+//                self.sendUpdated(scheduledActivities: [schedule])
+//
+//                // upload the archive
+//                SBBDataArchive.encryptAndUploadArchives([archive])
+//            }
+//        }
+//    }
+    
+    override func update(schedule: SBBScheduledActivity, task: ORKTask, result: ORKTaskResult, finishedOn: Date?) {
```

Comment:
This is never called. ResearchKit

Reply:
Hm, I wonder if that's why it's never updating the Journey view after you finish an activity until you quit and restart.